providers/freebsd/host_freebsd_cgo_test.go

```diff
@@ -0,0 +1,53 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build freebsd && cgo
+
+package freebsd
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/elastic/go-sysinfo/internal/registry"
+)
+
+var _ registry.HostProvider = freebsdSystem{}
+
+func TestHost(t *testing.T) {
```

Comment:
Can we test for the presence of some keys?

Reply:
At the moment, the only thing that is being tested without human intervention is that the calls return without error. The results could all be empty or non-sane, but the only way we can know this is if a human reads the test logs. We could probably do better at least by checking that some of the expected values (keys) are present.

Reply:
Understood. I added [additional validations](https://github.com/redanthrax/go-sysinfo/pull/1/files#diff-3fb6c9df800f5483fe1cbf78eba1efb733e4efbf8220168200c7a729fda73630L30) to the tests.

Extracted code from the [updated PR](https://github.com/redanthrax/go-sysinfo/pull/1):

```go
// TestArchitecture -- I can also validate with `uname -p` if you prefer
assert.Regexp(t, `(amd64|i386|powerpc(64(le)?|spe)?|armv(6|7)|aarch64|riscv64*|mips(n)?(32|64)?(el)?|sparc64)`, arch)

// TestKernelVersion
cmd := exec.Command("/bin/freebsd-version", "-r")

// TestMachineID -- I know Regex is slow here, but trying to avoid extra packages
assert.Regexp(t, "^[a-zA-Z0-9]{8}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{4}-[a-zA-Z0-9]{12}$", machineID)
```